### PR TITLE
feat: optimism mainnet support

### PR DESCRIPTION
**Pull request by ncitron**
- Added mainnet configs
- Added better timeout logic for engine api
- Increased cached unsafe blocks size

### CODE REVIEW

File: src/config/mod.rs [hunk 1]
```diff
@@ -204,6 +204,43 @@ impl Default for DefaultsProvider {
...
+                gas_limit: U256::from(30_000_000),
+                l1_fee_overhead: U256::from(188),
+                l1_fee_scalar: U256::from(684000),
+                unsafe_block_singer: addr("0xAAAA45d9549EDA09E70937013520214382Ffc4A2"),
+            },
+            batch_inbox: addr("0xff00000000000000000000000000000000000010"),
+            deposit_contract: addr("0xbEb5Fc579115071764c7423A4f12eDde41f106Ed"),
+            system_config_contract: addr("0x229047fed2591dbec1eF1118d64F7aF3dB9EB290"),
+            // TODO: replace with correct address
+            l2_to_l1_message_passer: addr("0x0000000000000000000000000000000000000000"),
```
Comment (clabby) [inline src/config/mod.rs:235]:
```suggestion
            l2_to_l1_message_passer: addr("0x4200000000000000000000000000000000000016"),
```

**Review by clabby — APPROVED** [review]
Values LGTM

File: src/config/mod.rs (hunk 1 above)
Comment (ncitron) [reply src/config/mod.rs:235]: Thanks!

**Comment by merklefruit** [comment]
LFGGG 🔥🔥

**Pull request merged**